CMakeLists.txt

```diff
@@ -155,6 +169,10 @@ endif()
 if(TARGET quickcpplib::hl)
   target_link_libraries(outcome_hl INTERFACE quickcpplib::hl)
 endif()
+if(TARGET status-code::hl)
+  target_link_libraries(outcome_hl INTERFACE status-code::hl)
+  all_compile_definitions(PUBLIC OUTCOME_USE_SYSTEM_STATUS_CODE=1)
```

Comment:
Macros in cmake should disable optional functionality rather than change semantics because a lot of users will use this library as header only without any cmake involved. That's probably the case here but ...